### PR TITLE
sys-apps/pnpm-bin: update to 9.5.0, fix distdir and remove the -bin

### PR DESCRIPTION
**Pull request by groche97**
The package is apparently abandoned, I update the binary changing this to pnpm instead of pnpm-bin in the system to be compatible with scripts, like rust-bin or other -bin packages. 
When I try to update the manifest doesn't change very well because it detects the old file in distfiles, I changed the download bin in distfiles to ${P} to fix it and simplify the update process.

### CODE REVIEW

**Review by antecrescent — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Have you thought about [requesting access to the server](https://wiki.gentoo.org/wiki/Project:GURU/Information_for_Contributors) and contributing directly, yet?
That way, your changes generally get merged quicker into `master` and it's less work for participants overall.

File: sys-apps/pnpm-bin/Manifest
[the remark this answers is not in the document]
Comment (antecrescent) [reply sys-apps/pnpm-bin/Manifest:1]: No need to open a new PR, just redo the changes and force-push e.g.

```
git reset --hard HEAD~2
pushd sys-apps/pnpm-bin
sed -e 's/1999-2023/2023-2024/' -e '/^SRC_URI/s/"$/ -> ${P}"/' \
-e 's/${PN}/pnpm/' -e '/newbin/s/pnpm-linux-x64/${P}/' \
-i pnpm-bin-9.0.6.ebuild
sed 's/pnpm-linux-x64/pnpm-bin-9.0.6/' -i Manifest
git add pnpm-bin-9.0.6.ebuild
pkgdev commit -e # edit the commit message to reflect your changes
cp pnpm-bin-9.{0.6,5.0}.ebuild
git add pnpm-bin-9.5.0.ebuild
pkgdev manifest
pkgdev commit
popd
git push -f

[the remark this answers is not in the document]
Comment (antecrescent) [reply sys-apps/pnpm-bin/Manifest:1]: Thanks _you_ for contributing ;) This PR will get closed automatically once e00f1d337d408a87c8c7843f511423f2064c19d1 gets merged into `master`.

**Comment by antecrescent** [comment]
@vitaly-zdanevich Do you still intend to maintain this package? If not, please remove yourself from `metadata.xml`.

**Comment by vitaly-zdanevich** [comment]
> @vitaly-zdanevich Do you still intend to maintain this package? If not, please remove yourself from `metadata.xml`.

I added this package because our https://complexnumbers.ru uses it, but I do not use pnpm every day..

**Pull request closed without merging**